demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -70,34 +97,248 @@
+ * broker.
+ */
+#define MQTT_KEEP_ALIVE_INTERVAL_SECONDS    ( 60U )
+
+/**
+ * @brief Delay between MQTT publishes in seconds.
+ */
+#define DELAY_BETWEEN_PUBLISHES_SECONDS     ( 1U )
+
+/**
+ * @brief Transport timeout in milliseconds for transport send and receive.
+ */
+#define TRANSPORT_SEND_RECV_TIMEOUT_MS      ( 100 )
+
+/*-----------------------------------------------------------*/
+
+/* @brief errno to check transport error. */
+extern int errno;
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief globalEntryTime entry time into the application to use as a reference
+ * timestamp in #getTimeMs function. #getTimeMs will always return the difference
+ * of current time with the globalEntryTime. This will reduce the chances of
+ * overflow for 32 bit unsigned integer used for holding the timestamp.
+ *
+ */
+static uint32_t globalEntryTimeMs = 0U;
+
+/**
+ * @brief Packet Identifier generated when Subscribe request was sent to the broker;
+ * it is used to match received Subscribe ACK to the transmitted subscribe.
+ */
+static uint16_t globalSubscribePacketIdentifier = 0U;
+
+/**
+ * @brief Packet Identifier generated when Unsubscribe request was sent to the broker;
+ * it is used to match received Unsubscribe response to the transmitted unsubscribe
+ * request.
+ */
+static uint16_t globalUnsubscribePacketIdentifier = 0U;
 
 /*-----------------------------------------------------------*/
 
 /**
- * @brief Establish a TCP connection to the given server.
+ * @brief Creates a TCP connection to the MQTT broker as specified by
+ * BROKER_ENDPOINT and BROKER_PORT defined at the top of this file.
  *
  * @param[in] pServer Host name of server.
  * @param[in] port Server port.
  *
- * @return A file descriptor representing the TCP socket; -1 on failure.
+ * @return EXIT_FAILURE on failure; EXIT_SUCCESS on success.
+ */
+static int connectToServer( const char * pServer,
+                            uint16_t port );
+
+/**
+ * @brief Set up a TLS connection over an existing TCP connection.
+ *
+ * @param[in] tcpSocket Existing TCP connection.
+ *
+ * @return An SSL connection context; NULL on failure.
+ */
+static SSL * tlsSetup( int tcpSocket );
+
+/**
+ * @brief The transport send function provided to the MQTT context.
+ *
+ * @param[in] pSslContext Pointer to SSL context.
+ * @param[in] pMessage Data to send.
+ * @param[in] bytesToSend Length of data to send.
+ *
+ * @return Number of bytes sent; negative value on error;
+ * 0 for timeout or 0 bytes sent.
+ */
+static int32_t transportSend( MQTTNetworkContext_t pSslContext,
+                              const void * pMessage,
+                              size_t bytesToSend );
+
+/**
+ * @brief The transport receive function provided to the MQTT context.
+ *
+ * @param[in] pSslContext Pointer to SSL context.
+ * @param[out] pBuffer Buffer for receiving data.
+ * @param[in] bytesToSend Size of pBuffer.
```

Comment:
Should be `bytesToRecv`. I think it's also like this in plaintext demo.

Reply:
Yea. Thanks. Fixed.